### ISSUE TITLE
[BUG] Clear AOE answers when changing devices

### ISSUE THREAD

**Issue by emyl3** [CDCgov/prime-simplereport#7722]
**Current behavior**
- Switching between devices with a different supported disease does not clear the AOE answers

To repro:
- Save AOE question answers for a single entry test with a HIV device
- Change the device to a non-HIV device
- Save the results
- Look in the DB and notice that HIV-related AOE answers are still there and saved

**Expected behavior**
- Clear AOE answers specific to that device's supported disease when switching to another device with a different supported disease

**Comment by mehansen** [comment]
@emyl3 already covered this in the ticket, just adding my test steps here as well. Mostly just want to emphasize that this gets sent to RS.

I tested this in `dev4`, filling out a test card for a multiplex device with respiratory symptoms and then switched it to an HIV device and submitted a negative result. We ended up saving a test event with a negative HIV result that still had the Covid symptom information included. The test event data is what is sent to RS, so they would get a message for HIV with potentially invalid symptomatic info included.
